MvvmCross/iOS/iOS/Views/Presenters/MvxIosViewPresenter.cs

```diff
@@ -111,48 +111,43 @@ public virtual void Show(IMvxIosView view, MvxViewModelRequest request)
             MvxRootPresentationAttribute attribute,
             MvxViewModelRequest request)
         {
+            // check if viewController is trying to initialize a navigation stack
+            if (attribute.WrapInNavigationController)
+            {
+                var navigationController = CreateNavigationController(viewController);
+                MasterNavigationController = navigationController as MvxNavigationController;
```

Comment:
Shouldn't we call `CloseMasterNavigationController` before assigning a new value to the `MasterNavigationController` property? If this is the case we should make the property readonly and add a set method which ensure everything is called in the correct order.

@nmilcoff could you maybe double check if my suggestion makes sense?